### PR TITLE
Added control of gain for each drum and a mix UI for gain and pan.

### PR DESCRIPTION
**Pull request by android-usb**

### CODE REVIEW

File: samples/drumthumper/src/main/java/com/google/oboe/sample/drumthumper/DrumThumperActivity.kt [hunk 1]
```diff
@@ -99,6 +103,26 @@ class DrumThumperActivity : AppCompatActivity(), TriggerPad.DrumPadTriggerListen
...
+    }
+
+    fun panPosToPanVal(pos: Int) : Float {
+        // map 0 -> 200 to -1.0f -> 1..0f
+        return (pos.toFloat() - 100.0f) / 100.0f
+    }
+
+    fun panValToPanPos(value: Float) : Int {
+        // map -1.0f -> 1.0f to 0 -> 200
+        return ((value * 200.0f) + 100.0f).toInt()
```
Comment (philburk) [inline samples/drumthumper/src/main/java/com/google/oboe/sample/drumthumper/DrumThumperActivity.kt:123]: Not inverse of panValToPanPos()

File: samples/drumthumper/src/main/java/com/google/oboe/sample/drumthumper/DrumThumperActivity.kt [hunk 2]
```diff
@@ -125,46 +149,79 @@ class DrumThumperActivity : AppCompatActivity(), TriggerPad.DrumPadTriggerListen
...
-        run {
-            var pad: TriggerPad = findViewById(R.id.crashPad)
-            pad.addListener(this)
-        }
-
+        var sb : SeekBar;
+
+        // "Kick" drum
+        (findViewById(R.id.kickPad) as TriggerPad).addListener(this)
+        sb = (findViewById(R.id.kickPan) as SeekBar)
```
Comment (philburk) [inline samples/drumthumper/src/main/java/com/google/oboe/sample/drumthumper/DrumThumperActivity.kt:156]: Can this be a method?

File: samples/drumthumper/src/main/cpp/DrumPlayerJNI.cpp [hunk 3]
```diff
@@ -128,6 +128,27 @@ JNIEXPORT void JNICALL Java_com_google_oboe_sample_drumthumper_DrumPlayer_restar
...
 }
 
+JNIEXPORT void JNICALL Java_com_google_oboe_sample_drumthumper_DrumPlayer_setPan(
+        JNIEnv *env, jobject thiz, jint index, jfloat pan) {
+    sDTPlayer.setPan(index, pan);
+}
+
+JNIEXPORT jfloat JNICALL Java_com_google_oboe_sample_drumthumper_DrumPlayer_getPan(
+        JNIEnv *env, jobject thiz, jint  index) {
+    // TODO: implement getPan()
```
Comment (dturner) [inline samples/drumthumper/src/main/cpp/DrumPlayerJNI.cpp:138]: Can remove this TODO if you implemented `getPan`

File: samples/drumthumper/src/main/java/com/google/oboe/sample/drumthumper/DrumThumperActivity.kt [hunk 4]
```diff
@@ -99,6 +103,26 @@ class DrumThumperActivity : AppCompatActivity(), TriggerPad.DrumPadTriggerListen
         }
     }
 
+    // UI Helpers
+    fun gainPosToGainVal(pos: Int) : Float {
+        // map 0 -> 200 to 0.0f -> 2.0f
+        return pos.toFloat() / 100.0f
```
Comment (dturner) [inline samples/drumthumper/src/main/java/com/google/oboe/sample/drumthumper/DrumThumperActivity.kt:109]: Magic numbers should be moved to constants with descriptive names

File: samples/drumthumper/src/main/java/com/google/oboe/sample/drumthumper/DrumThumperActivity.kt [hunk 5]
```diff
@@ -125,46 +149,79 @@ class DrumThumperActivity : AppCompatActivity(), TriggerPad.DrumPadTriggerListen
...
-
-        run {
-            var pad: TriggerPad = findViewById(R.id.crashPad)
-            pad.addListener(this)
-        }
-
+        var sb : SeekBar;
+
+        // "Kick" drum
+        (findViewById(R.id.kickPad) as TriggerPad).addListener(this)
```
Comment (dturner) [inline samples/drumthumper/src/main/java/com/google/oboe/sample/drumthumper/DrumThumperActivity.kt:155]: If you use view bindings you can avoid using `findViewById`: https://developer.android.com/topic/libraries/view-binding

File: samples/drumthumper/src/main/java/com/google/oboe/sample/drumthumper/DrumThumperActivity.kt [hunk 6]
```diff
@@ -125,46 +149,79 @@ class DrumThumperActivity : AppCompatActivity(), TriggerPad.DrumPadTriggerListen
...
-            var pad: TriggerPad = findViewById(R.id.ridePad)
-            pad.addListener(this)
-        }
-
-        run {
-            var pad: TriggerPad = findViewById(R.id.crashPad)
-            pad.addListener(this)
-        }
-
+        var sb : SeekBar;
```
Comment (dturner) [inline samples/drumthumper/src/main/java/com/google/oboe/sample/drumthumper/DrumThumperActivity.kt:152]: consider `gainSeekBar` or something more descriptive as a variable name

File: samples/drumthumper/src/main/java/com/google/oboe/sample/drumthumper/DrumThumperActivity.kt (hunk 1 above)
Comment (android-usb) [reply samples/drumthumper/src/main/java/com/google/oboe/sample/drumthumper/DrumThumperActivity.kt:123]: Fixed.

File: samples/drumthumper/src/main/cpp/DrumPlayerJNI.cpp (hunk 3 above)
Comment (android-usb) [reply samples/drumthumper/src/main/cpp/DrumPlayerJNI.cpp:138]: Done

File: samples/drumthumper/src/main/java/com/google/oboe/sample/drumthumper/DrumThumperActivity.kt (hunk 2 above)
Comment (android-usb) [reply samples/drumthumper/src/main/java/com/google/oboe/sample/drumthumper/DrumThumperActivity.kt:156]: Done.

File: samples/drumthumper/src/main/java/com/google/oboe/sample/drumthumper/DrumThumperActivity.kt (hunk 5 above)
Comment (android-usb) [reply samples/drumthumper/src/main/java/com/google/oboe/sample/drumthumper/DrumThumperActivity.kt:155]: I will investigate.

File: samples/drumthumper/src/main/java/com/google/oboe/sample/drumthumper/DrumThumperActivity.kt (hunk 6 above)
Comment (android-usb) [reply samples/drumthumper/src/main/java/com/google/oboe/sample/drumthumper/DrumThumperActivity.kt:152]: done.

File: samples/drumthumper/src/main/java/com/google/oboe/sample/drumthumper/DrumThumperActivity.kt (hunk 4 above)
Comment (android-usb) [reply samples/drumthumper/src/main/java/com/google/oboe/sample/drumthumper/DrumThumperActivity.kt:109]: Done.

File: samples/drumthumper/src/main/java/com/google/oboe/sample/drumthumper/DrumThumperActivity.kt (hunk 5 above)
Comment (android-usb) [reply samples/drumthumper/src/main/java/com/google/oboe/sample/drumthumper/DrumThumperActivity.kt]: I'm handling it with findViewById<TriggerPad>(....). I'd rather not tackle bindings right now.

**Pull request merged**